openml/datasets/functions.py

```diff
@@ -383,21 +386,42 @@ def get_dataset(
         If no version is specified, retrieve the least recent still active version.
     error_if_multiple : bool (default=False)
         If ``True`` raise an error if multiple datasets are found with matching criteria.
-    cache_format : str (default='pickle')
+    cache_format : str (default='pickle') in {'pickle', 'feather'}
         Format for caching the dataset - may be feather or pickle
         Note that the default 'pickle' option may load slower than feather when
         no.of.rows is very high.
     download_qualities : bool (default=True)
         Option to download 'qualities' meta-data in addition to the minimal dataset description.
+        If True, download and cache the qualities file.
+        If False, create the OpenMLDataset without qualities metadata. The data may later be added
+        to the OpenMLDataset hrough the `OpenMLDataset.load_metadata(qualities=True)` method.
+    download_features_meta_data : bool (default=True)
+        Option to download 'features' meta-data in addition to the minimal dataset description.
+        If True, download and cache the features file.
+        If False, create the OpenMLDataset without features metadata. The data may later be added
+        to the OpenMLDataset through the `OpenMLDataset.load_metadata(features=True)` method.
     download_all_files: bool (default=False)
         EXPERIMENTAL. Download all files related to the dataset that reside on the server.
         Useful for datasets which refer to auxiliary files (e.g., meta-album).
+    force_refresh_cache : bool (default=False)
+        Force the cache to refreshed by deleting the cache directory and re-downloading the data.
+        Note, if `force_refresh_cache` is True, `get_dataset` is NOT thread/multiprocessing safe,
+        because this creates a race condition to creating and deleting the cache; as in general with
+        the cache.
 
     Returns
     -------
     dataset : :class:`openml.OpenMLDataset`
         The downloaded dataset.
     """
+    if any([download_qualities, download_features_meta_data]):
+        warnings.warn(
```

Comment:
I would suggest approaching this a bit differently:
1. We set the flag to `None` above and emit a `DeprecationWarning` then.
2. We ask the user to pass a `bool` to disable this warning.
3. We internally assume `True` if `None` is given.
4. And also do this for `download_dataset`.

Reply:
Your idea sounds much better. I will change this accordingly.

One question, I noticed that Python, by default, suppresses deprecation warnings. Should we maybe switch to user warnings? Otherwise, users only see the warning if they enable deprecation warnings. 

We could also change the warning level in our code, but this would change settings in the outer scope unless we clean up properly.

Reply:
I added your suggestion and switched to user warnings for now.

Reply:
scikit-learn appears to use `FutureWarning`, so we might use that instead?